components/hab/src/command/pkg/binlink.rs

```diff
@@ -126,9 +141,105 @@ where
     Ok(())
 }
 
+fn is_executable(extension: Option<&OsStr>) -> bool {
+    match extension {
+        Some(ext) => match env::var_os("PATHEXT") {
+            Some(val) => env::split_paths(&val).any(|e| {
+                e.to_string_lossy() == format!(".{}", ext.to_string_lossy().to_uppercase())
+            }),
+            None => false,
+        },
+        None => false,
+    }
+}
+
+fn is_dest_on_path<T: AsRef<Path>>(dest_dir: T) -> bool {
+    if let Some(val) = env::var_os("PATH") {
+        env::split_paths(&val).any(|p| p == dest_dir.as_ref())
+    } else {
+        false
+    }
+}
+
+struct Binlink {
+    path: PathBuf,
+    src: PathBuf,
+}
+
+#[cfg(not(target_os = "windows"))]
+impl Binlink {
+    pub fn new<T: AsRef<Path>>(src: T, dest_dir: T) -> Self {
+        Binlink {
+            path: dest_dir.as_ref().join(
+                src.as_ref()
+                    .file_name()
+                    .expect("Binlink source path does not resolve to a file"),
```

Comment:
I figured the `expect` was ok here since `src` is coming from `find_command_in_pkg` which would never return a non-file. Do you think there is still merit in a `result`?

Reply:
I agree that I don't see a way that it could happen now, but there's no guarantee based on `Binlink`'s own handling of the source input that it won't end in `..`. A new caller or a change in `find_command_in_pkg` could cause a panic that I think would better be avoided.